Chessington.GameEngine/Pieces/Moves.cs

```diff
@@ -1,49 +1,44 @@
-﻿using System.Collections.Generic;
-using System.Linq;
-
-namespace Chessington.GameEngine.Pieces {
-    public class Moves {
-        public static IEnumerable<Square> GetLateralMoves(Board board, Square currentPosition) {
+﻿using System;
+using System.Collections.Generic;
+
+namespace Chessington.GameEngine.Pieces
+{
+    public class Moves
+    {
+        public static List<Square> GetBeamMoves(Board board, Square currentPosition, 
+            Player currentPlayer, Func<Square, Square> iterator) {
             List<Square> availableMoves = new List<Square>();
-            foreach (var index in Enumerable.Range(0, GameSettings.BoardSize)) {
-                if (currentPosition.Col != index) {
-                    availableMoves.Add(new Square(currentPosition.Row, index));
-                }
-
-                if (currentPosition.Row != index) {
-                    availableMoves.Add(new Square(index, currentPosition.Col));
-                }
+            while ((currentPosition = iterator.Invoke(currentPosition)).IsValid()) {
+                if (board.IsEmptyOrOpponent(currentPosition, currentPlayer)) availableMoves.Add(currentPosition);
```

Comment:
please add curly brackets after if statement even if it's a one liner, it is more readable that way.